linux/hci/hci.go

```diff
@@ -841,6 +841,17 @@ func (h *HCI) handleDisconnectionComplete(b []byte) error {
 }
 
 func (h *HCI) handleEncryptionChange(b []byte) error {
+	e := evt.EncryptionChange(b)
+	h.muConns.Lock()
+	defer h.muConns.Unlock()
```

Comment:
I think holding the lock for the whole function is the safer thing to do.

If we get an async disconnect this would ensure we get "encryption failed", then "disconnected", though in all reality this is reading from a socket so unlikely we'd have a scenario that the ordering would get swapped.